### PR TITLE
feat(manifest): add allow source IP field

### LINKED ISSUE #1234: Add IP Allow List to LoadBalancedWeb Service Pattern
There are lots of occasions where you might want to limit the folks who can access their load balanced web service based on IP address range.

This is something you can do using [Source IP Conditions](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#source-ip-conditions) in our listener configuration. 

We might consider updating our manifest to look like:

```yaml

name: my-restricted-service
# The "architecture" of the service you're running.
type: Load Balanced Web Service

image:
  # Path to your service's Dockerfile.
  build:./Dockerfile
  # Port exposed through your container to route traffic to it.
  port: 3030

http:
  # Requests to this path will be forwarded to your service. 
  # To match all requests you can use the "/" path. 
  path: /
  # Only allow callers from this list of IP addresses
  allow-list:
    - 203.0.113.0/24
```

### PR DESCRIPTION
**Pull request by huanjani**
This gives users the option to limit access to their load balanced web services to certain IP address ranges.

Resolves https://github.com/aws/copilot-cli/issues/1234; partially addresses https://github.com/aws/copilot-cli/issues/1453.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/workloads/services/lb-web/cf.yml [hunk 1]
```diff
@@ -246,6 +246,14 @@ Resources:
...
           Type: forward
       Conditions:
+      {{- if .AllowedSourceIps}}
+        - Field: 'source-ip'
+          SourceIpConfig:
+            Values:
+            {{- range $sourceIP := .AllowedSourceIps}}
+            - {{$sourceIP}}
+            {{- end}}
+            {{- end}}
```
Comment (iamhopaul123) [inline templates/workloads/services/lb-web/cf.yml]: Could you make the last `end` to be aligned with the `if` above?

**Review by iamhopaul123 — APPROVED** [review]
LGTM with just one small nit :shipit:

File: templates/workloads/services/lb-web/cf.yml (hunk 1 above)
Comment (huanjani) [reply templates/workloads/services/lb-web/cf.yml]: Yes! Thanks!

File: site/content/docs/manifest/lb-web-service.md [hunk 2]
```diff
@@ -128,12 +128,15 @@ The approximate amount of time, in seconds, between health checks of an individu
...
  
-<span class="parent-field">http.</span><a id="http-target_container" href="#http-target_container" class="field">`targetContainer`</a> <span class="type">String</span>  
+<span class="parent-field">http.</span><a id="http-target-container" href="#http-target-container" class="field">`target_container`</a> <span class="type">String</span>  
 A sidecar container that takes the place of a service container.
                                  
 <span class="parent-field">http.</span><a id="http-stickiness" href="#http-stickiness" class="field">`stickiness`</a> <span class="type">Boolean</span>  
 Indicates whether sticky sessions are enabled.
 
+<span class="parent-field">http.</span><a id="http-allowed-source-ips" href="#http-allowed-source-ips" class="field">`allowed_source_ips`</a> <span class="type">Array of Strings</span>  
+CIDR IP addresses permitted to access your service.
```
Comment (efekarakus) [inline site/content/docs/manifest/lb-web-service.md:138]: nit: Can we add a small example here to illustrate how to use it:
```yaml
http:
   allowed_source_ips: ["192.0.2.0/24", "198.51.100.10/32"]
```

**Review by efekarakus — APPROVED** [review]
Just to double check, we're not writing anything in the manifest to showcase this field right?

I'm fine with that I just wanted to make sure.

File: site/content/docs/manifest/lb-web-service.md (hunk 2 above)
Comment (huanjani) [reply site/content/docs/manifest/lb-web-service.md:138]: yes! done.

**Comment by huanjani** [comment]
> Just to double check, we're not writing anything in the manifest to showcase this field right?
> 
> I'm fine with that I just wanted to make sure.

correct!

**Pull request merged**